src/api/procedures/modifyMultiSig.ts

```diff
@@ -1,101 +1,141 @@
     context,
   } = this;
   const { signers, multiSig: multi } = args;
 
-  // ensure calling account is creator, other actions require a proposal from the multisig
-  const signingIdentity = await context.getSigningIdentity();
-  const creator = await multi.getCreator();
-  if (!creator.isEqual(signingIdentity)) {
+  const [signingIdentity, creator] = await Promise.all([
+    context.getSigningIdentity(),
+    multi.getCreator(),
+  ]);
+
+  if (signersToAdd.length === 0 && signersToRemove.length === 0) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
-      message: 'A MultiSig can only be modified by its creator',
+      message:
+        'The given signers are equal to the current signers. At least one signer should be added or removed',
     });
   }
 
-  const rawAddress = stringToAccountId(multi.address, context);
-  const { signers: currentSigners, signaturesRequired } = await multi.details();
-
-  if (signaturesRequired.gt(signers.length)) {
+  if (!creator.isEqual(signingIdentity)) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
-      message: 'The number of signatures required should not exceed the number of signers',
+      message: 'A MultiSig can only be modified by its creator',
     });
   }
 
-  const [addedSigners, removedSigners] = calculateSignerDelta(currentSigners, signers);
+  const rawAddress = stringToAccountId(multi.address, context);
 
-  if (addedSigners.length === 0 && removedSigners.length === 0) {
+  if (requiredSignatures.gt(signers.length)) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
-      message: 'The given signers are equal to the current signers',
+      message: 'The number of required signatures should not exceed the number of signers',
     });
   }
 
-  if (addedSigners.length > 0) {
-    const rawAddedSigners = addedSigners.map(s => signerToSignatory(s, context));
-    this.addTransaction({
-      transaction: tx.multiSig.addMultisigSignersViaCreator,
-      args: [rawAddress, rawAddedSigners],
-    });
+  const transactions = [];
+
+  if (signersToAdd.length > 0) {
+    const rawAddedSigners = signersToAdd.map(signer => signerToSignatory(signer, context));
+    transactions.push(
+      checkTxType({
+        transaction: tx.multiSig.addMultisigSignersViaCreator,
+        args: [rawAddress, rawAddedSigners],
+      })
+    );
   }
 
-  if (removedSigners.length > 0) {
-    const rawRemovedSigners = removedSigners.map(s => signerToSignatory(s, context));
-    this.addTransaction({
-      transaction: tx.multiSig.removeMultisigSignersViaCreator,
-      args: [rawAddress, rawRemovedSigners],
-    });
+  if (signersToRemove.length > 0) {
+    const rawRemovedSigners = signersToRemove.map(signer => signerToSignatory(signer, context));
+    transactions.push(
+      checkTxType({
+        transaction: tx.multiSig.removeMultisigSignersViaCreator,
+        args: [rawAddress, rawRemovedSigners],
+      })
+    );
   }
```

Comment:
These two blocks look similar enough that they might be refactored into a function